packages/@aws-cdk/assert/lib/assertions/have-output.ts

```diff
@@ -0,0 +1,82 @@
+import {Assertion} from "../assertion";
+import {StackInspector} from "../inspector";
+
+class HaveOutputAssertion extends Assertion<StackInspector> {
+  constructor(private readonly outputName?: string, private readonly exportName?: string, private outputValue?: any) {
```

Comment:
`exportName` will, in many situations, be an intrinsic, as they need to be unique for a given account/region pair. A recommended pattern for naming exports for example is (read as the YAML form): `Fn::Sub: ${AWS::StackName}-OutputName`

I feel like requiring it to be provided as a `string` is not ideal here, for this very reason.

Reply:
This also lacks TSDoc comments.

Reply:
Changed exportName to `any` type, refactored code to accommodate for that change.
In regards to the lack of TSDoc, I feel like we don't need to duplicate TSDocs from `haveOutput` function in the same file. Also I think `haveOutput` is the true interface for the user of that assertion, not the class itself.

Reply:
@RomainMuller , also regarding 
> I wonder if there's any value in allowing `outputName` to be a `RegExp` so we can pattern-match it
> I wonder if there's any value in checking for absense of an `exportName`
I see your point and those suggestions make sense. However it does not align with the approach I was taking (driven by use-case). I guess it requires further analysis and clearer use-cases. For example, if we make `outputName` of a `RegExp` type, `exportName` would be expected to correspondingly accept `RegExp` as well. But that is already string-value expression
Should we just move those two items into a separate issue and later come up with clearer scenarios?